extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/params/HttpRequestParams.java

```diff
@@ -67,6 +70,38 @@ public Request toRequest(@Nullable Supplier<InputStream> bodySupplier) {
         return requestBuilder.build();
     }
 
+    public String getMethod() {
+        return method;
+    }
+
+    public String getBaseUrl() {
+        return baseUrl;
+    }
+
+    public String getPath() {
+        return path;
+    }
+
+    public String getQueryParams() {
+        return queryParams;
+    }
+
+    public String getContentType() {
+        return contentType;
+    }
+
+    public String getBody() {
+        return body;
+    }
+
+    public boolean isNonChunkedTransfer() {
+        return nonChunkedTransfer;
+    }
+
+    public Map<String, String> getHeaders() {
+        return headers;
```

Comment:
Do we really need to expose all these getters?

Reply:
I rowed against my own belief to avoid introducting methods for test's sake, but it was the only way to have them tested up properly, probably the design here of having a data class (HttpRequestParams) to have knowledge about the okhttp.Request could be considered not that good, maybe in a next refactoring the construction of an okhttp request from the params could be moved to another component (like a factory), leaving the HttpRequestParams as a simple data class.

Just thinking out loud... in fact it could be considered an EDC abstraction of the http request and, why not, moved to the `http-spi` and used in the `EdcHttpClient`, but maybe this is too far for the moment.